tests/handlers/test_federation.py

```diff
@@ -190,6 +203,145 @@ def test_rejected_state_event_state(self):
+
+        # build a floating outlier member state event
+        fake_prev_event_id = "$" + random_string(43)
+        member_event = event_from_pdu_json(
+            {
+                "type": EventTypes.Member,
+                "content": {
+                    "membership": "join",
+                },
+                "state_key": OTHER_USER,
+                "room_id": room_id,
+                "sender": OTHER_USER,
+                "depth": most_recent_prev_event_depth,
+                "prev_events": [fake_prev_event_id],
+                "auth_events": auth_event_ids.copy(),
+                "origin_server_ts": self.clock.time_msec(),
+                "signatures": {
+                    OTHER_SERVER: {"ed25519:key_version": "SomeSignatureHere"}
+                },
+            },
+            room_version,
+            outlier=True,
+        )
+        logger.error("member_event event_id=%s", member_event.event_id)
+        logger.error(
+            "member_event before event_id=%s auth_events=%s",
+            member_event.event_id,
+            member_event.auth_events,
+        )
+        filter_auth_events_on_event(member_event, auth_events)
+        logger.error("member_event after auth_events=%s", member_event.auth_events)
+
+        # build and send an event authed based on the member event
+        raw_auth_events = auth_events + [member_event]
+        message_event = event_from_pdu_json(
+            {
+                "type": EventTypes.Message,
+                "content": {},
+                "room_id": room_id,
+                "sender": OTHER_USER,
+                "depth": most_recent_prev_event_depth,
+                "prev_events": prev_event_ids.copy(),
+                "auth_events": [ev.event_id for ev in raw_auth_events],
+                "origin_server_ts": self.clock.time_msec(),
+                "signatures": {
+                    OTHER_SERVER: {"ed25519:key_version": "SomeSignatureHere"}
+                },
+            },
+            room_version,
+        )
+        logger.error("message_event event_id=%s", message_event.event_id)
+        logger.error("message_event before auth_events=%s", message_event.auth_events)
+        filter_auth_events_on_event(message_event, raw_auth_events.copy())
+        logger.error("message_event after auth_events=%s", message_event.auth_events)
+
+        # Stub the /event_auth response from the OTHER_SERVER
+        async def get_event_auth(destination, room_id, event_id):
+            return [
+                auth_event
+                for auth_event in raw_auth_events
+                if auth_event.type is EventTypes.Create
+                or auth_event.type is EventTypes.PowerLevels
+                or auth_event.type is EventTypes.JoinRules
+                or (
+                    auth_event.type is EventTypes.Member
+                    and auth_event.state_key == message_event.sender
+                )
+            ]
```

Comment:
I used to have this derive directly from the events `auth_events` but [`GET /_matrix/federation/v1/event_auth/{roomId}/{eventId}`](https://matrix.org/docs/spec/server_server/r0.1.4#get-matrix-federation-v1-event-auth-roomid-eventid) returns different types of events from `_event_auth_handler.compute_auth_events(...)`.

[`GET /_matrix/federation/v1/event_auth/{roomId}/{eventId}`](https://matrix.org/docs/spec/server_server/r0.1.4#get-matrix-federation-v1-event-auth-roomid-eventid) returns `m.room.create`, `m.room.power_levels`, `m.room.join_rules`, and `m.room.member` events

Whereas `compute_auth_events` will only return `m.room.create`, `m.room.power_levels`, and `m.room.member` (missing the join rules)


Previous implementation (we only access for `message_event`):

```python
# Stub the /event_auth response from the OTHER_SERVER
async def get_event_auth(destination, room_id, event_id):
    logger.error("get_event_auth event_id=%s", event_id)
    event_map = {
        member_event.event_id: member_event,
        message_event.event_id: message_event,
    }
    # Convert the message auth event ID's back into full fledged events
    return [
        auth_event
        for auth_event in raw_auth_events
        if auth_event.event_id in event_map[event_id].auth_events
    ]
```

Reply:
`/event_auth` returns the full auth chain, rather  than just the auth events for the requested event, i.e it includes the auth events of the auth events recursively, which is why there is a difference.

Reply:
I was able to get away with making this much simpler 🎉 

```python
        # Stub the /event_auth response from the OTHER_SERVER
        async def get_event_auth(
            destination: str, room_id: str, event_id: str
        ) -> List[EventBase]:
            return auth_events
```